src/libasr/pass/pass_array_by_data.cpp

```diff
@@ -644,11 +644,29 @@ class RemoveArrayByDescriptorProceduresVisitor : public PassUtils::PassVisitor<R
             std::vector<std::string> to_be_erased;
 
             for( auto& item: current_scope->get_scope() ) {
-                if( v.proc2newproc.find(item.second) != v.proc2newproc.end() &&
-                    not_to_be_erased.find(item.second) == not_to_be_erased.end() ) {
+                if (ASR::is_a<ASR::Function_t>(*item.second)) {
+                    ASR::FunctionType_t* func_type = ASRUtils::get_FunctionType(item.second);
+                    if (func_type->m_arg_types[0] && ASR::is_a<ASR::Class_t>(*func_type->m_arg_types[0])) {
```

Comment:
You are accessing a dangling pointer, you have to do it like this:
```suggestion
                    if (func_type->n_arg_types >= 1 && ASR::is_a<ASR::Class_t>(*func_type->m_arg_types[0])) {
```